lib/IRGen/GenClangType.cpp

```diff
@@ -594,7 +597,9 @@ clang::CanQualType GenClangType::visitSILFunctionType(CanSILFunctionType type) {
       break;
 
     case ParameterConvention::Direct_Owned:
-      llvm_unreachable("block takes owned parameter");
+      extParamInfo = extParamInfo.withIsConsumed(true);
+      break;
```

Comment:
@varungandhi-apple You'll need to make an analogous change.

Reply:
(Comment for my own reference): https://github.com/apple/swift/pull/27479 needs to have an analogous change in `ClangTypeConverter.cpp`.

Reply:
Actually, we don't need to make any changes since `visitSILFunctionType` in `ClangTypeConverter` is unreachable -- it should receive only AST types.

Reply:
`owned` can appear in AST function parameters as well; it's in the `ParameterTypeFlags`.